src/main/scala/akka/persistence/cassandra/query/journal/EventsByTagPublisher.scala

```diff
@@ -0,0 +1,92 @@
+package akka.persistence.cassandra.query.journal
+
+import java.nio.ByteBuffer
+import java.util.UUID
+
+import akka.actor.Props
+import akka.persistence.PersistentRepr
+import akka.persistence.query.EventEnvelope
+import akka.serialization.{Serialization, SerializationExtension}
+import com.datastax.driver.core.{Row, Session}
+import com.datastax.driver.core.utils.{Bytes, UUIDs}
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.collection.JavaConverters._
+
+import akka.persistence.cassandra.listenableFutureToFuture
+import EventsByTagPublisher._
+
+private[journal] object EventsByTagPublisher {
```

Comment:
Not tested yet so there will be issues. But this is conceptually how I would think EventsByTagPublisher should look. Again actor fetching the data would be completely fine. Subpartitioning is not considered at the moment.